package.json

```diff
@@ -2,14 +2,18 @@
   "name": "@storyous/common-utils",
   "version": "14.1.0",
   "description": "Common utils for storyous microservices",
-  "main": "lib/index.js",
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
   "bin": {
     "download-translations": "./scripts/download-translations.js"
   },
   "scripts": {
-    "test": "npm run test:lint && npm run test:mocha",
-    "test:mocha": "mocha test",
-    "test:lint": "eslint lib"
+    "test": "run-p test:lint build --aggregate-output && npm run test:mocha",
+    "test:mocha": "cd test && mocha ./",
+    "test:lint": "eslint lib",
+    "dev": "tsc --watch",
+    "build": "rm -rf dist && tsc",
```

Comment:
run-p nebo rm jsou příkazy, který nepoběží na windows. Jestli jsem teda jedinej s windows, tak si to upravím až to budu potřebovat, ale jestli je tu víc lidí nebo pokud to chceme mít nezávislý na OS, pak by se to mělo nahradit něčím co není OS-specific.

Reply:
btw: tohle je nastavení ze starýho projektu

```
scripts": {
    "start": "cross-env-shell CUSTOM_MAX_OLD_SPACE_SIZE?=460 node --max-old-space-size=${CUSTOM_MAX_OLD_SPACE_SIZE} dist/index.js",
    "start-win": "node --max-old-space-size=460 dist/index.js",
    "start-watch": "npm run build && npm run watch",
    "build": "npm run build-ts",
    "build-ts": "tsc",
    "tslint": "tslint -c tslint.json -p tsconfig.json -t verbose",
    "tslint-fix": "tslint --fix -c tslint.json -p tsconfig.json -t verbose",
    "watch": "concurrently -k -p \"[{name}]\" -n \"TypeScript,Node\" -c \"yellow.bold,cyan.bold,green.bold\" \"npm run watch-ts\" \"npm run serve\"",
    "watch-ts": "tsc -w",
    "serve": "nodemon dist/index.js",
    "docker-build": "docker build  -t massive/subscription-management .",
    "docker-run": "docker run -dP -e \"CONFIG_ENV=localhost\" massive/subscription-management",
    "docker-compose-up": "docker-compose up",
    "mocha-test": "cross-env-shell MOCHA_REPORTER?=spec mocha --timeout 15000 --exit --reporter ${MOCHA_REPORTER} --require ts-node/register -P tsconfig.json",
    "test-unit": "npm run mocha-test -- \"test/unit/*.spec.ts\"",
    "test-component": "npm run mocha-test -- \"test/component/*.spec.ts\"",
    "test-integration": "npm run mocha-test -- \"test/integration/*.spec.ts\"",
    "pretest": "shx rm -rf coverage",
    "pretest-all": "npm run build && npm run tslint",
    "test-all": "npm run test-unit && npm run test-component",
    "test": "nyc --check-coverage --lines 80 npm run test-all",
    "mocha-bamboo-test": "cross-env-shell MOCHA_REPORTER=mocha-bamboo-reporter npm run test || true",
    "start-validate-job": "node dist/jobs/validator.js",
    "start-cancel-zuora-job": "node dist/jobs/cancelZuora.js",
    "start-cancel-salesforce-job": "node dist/jobs/cancelSalesforce.js"
  },

  "dependencies": {
    "@perform/middleware-common-endpoints": "^1.0.4",
    "@perform/middleware-common-trackers": "^1.0.2",
    "axios": "^0.18.1",
    "axios-mock-adapter": "^1.17.0",
    "axios-retry": "^3.1.2",
    "body-parser": "^1.19.0",
    "bunyan": "^1.8.12",
    "compression": "^1.7.4",
    "cross-env-default": "^5.1.3-1",
    "express": "^4.17.1",
    "joi": "^14.3.1",
    "jsforce": "github:getndazn/jsforce-forked#patch-1",
    "lodash": "^4.17.15",
    "mongoose": "^5.7.12",
    "newrelic": "^5.13.1",
    "pkginfo": "^0.4.1",
    "squel": "^5.13.0"
  },
  "devDependencies": {
    "@perform/tslint-config": "^1.0.3",
    "@types/bunyan": "^1.8.6",
    "@types/chai": "4.1.7",
    "@types/express": "^4.17.2",
    "@types/joi": "^13.6.3",
    "@types/lodash": "^4.14.149",
    "@types/mocha": "^5.2.7",
    "@types/mongoose": "^5.5.32",
    "@types/node": "^10.17.5",
    "chai": "^4.2.0",
    "chai-as-promised": "^7.1.1",
    "chance": "^1.1.3",
    "concurrently": "^3.6.1",
    "mocha": "^5.2.0",
    "mocha-bamboo-reporter": "^1.1.2",
    "nodemon": "^1.19.4",
    "nyc": "^14.1.1",
    "shx": "^0.3.2",
    "sinon": "^6.3.5",
    "supertest": "^3.4.2",
    "ts-node": "^6.2.0",
    "tslint": "5.18.0",
    "typescript": "^3.6.4"
  }
```

A tohle v souboru tsconfig.json

```
{
  "compilerOptions": {
      "module": "commonjs",
      "target": "es2017",
      "moduleResolution": "node",
      "sourceMap": true,
      "outDir": "dist",
      "baseUrl": ".",
      "paths": {
          "*": [
              "node_modules/*",
              "src/types/*"
          ]
      }
  },
  "include": [
      "src/**/*"
  ]
} 
```

a .nycrc

```
{
    "include": [
        "src/**/*.ts"
    ],
    "exclude": [
        "data/**/*",
        "src/**/*.d.ts",
        "src/config/env/*.ts",
        "src/newrelic.ts",
        "src/index.ts",
        "src/jobs/*",
        "src/enums",
        "src/errors",
        "src/errors/*",
        "src/config/*",
        "src/components/salesforceApi/xmlBuilder",
        "src/components/**/remote/mock",
        "src/components/massLog/*.ts",
        "src/futureModules/serializers/*.ts"
    ],
    "all": true,
    "extension": [
        ".ts"
    ],
    "reporter": [
        "lcov",
        "text-summary",
        "clover"
    ]
}
```

Reply:
taky jedu na windows :)
to run-p zprostredkovava modul npm-run-all, ktery je jako devDependence, to rm- rf jsem si neuvedomil, nahradil jsem za rimraf (nyni take devDependence)
konfiguraky - super, je neco co bys chtel rovnou pridat?

Reply:
Jenom kdyby něco nefungovalo nebo jako inspiraci